server/schedule/filter/filters.go

```diff
@@ -543,6 +543,41 @@ func (f *ruleFitFilter) Target(opt opt.Options, store *core.StoreInfo) bool {
 	return placement.CompareRegionFit(f.oldFit, newFit) <= 0
 }
 
+type engineFilter struct {
+	scope      string
+	constraint placement.LabelConstraint
+}
+
+// NewEngineFilter creates a filter that filters store for special engine.
+func NewEngineFilter(scope string, allowUses ...string) Filter {
```

Comment:
Can constraint.Values be specified directly in the constructor?

Reply:
I have adds some comments.